lib/ansible/modules/system/mdadm.py

```diff
@@ -0,0 +1,382 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# (c) 2015, Adam Hamsik <haaaad@gmail.com>
+#
+# This file is part of Ansible
+#
+# This module is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This software is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this software.  If not, see <http://www.gnu.org/licenses/>.
+
+ANSIBLE_METADATA = {'status': ['preview'],
+                    'supported_by': 'community',
+                    'version': '1.0'}
+
+DOCUMENTATION = '''
+---
+module: mdadm
+version_added: 2.3
+short_description: Initialize and manage linux mdadm array
+description: Create, Manage linux software raid
+requirements: [ python json ]
+options:
+  executable:
+    description:
+        - Path to mdadm binary
+    default: mdadm
+  state:
+    description:
+      - Defines action which can be either RAID creation or removal. This will assemble
```

Comment:
I am a bit surprised, since "state" is describing action. That sound a bit weird to me to mix 2 differents concepts in one single argument.